Sources/Mailgun/Mailgun.swift

```diff
@@ -1,62 +1,126 @@
 import Vapor
 import Foundation
 
+
+// MARK: - Service
+
 public protocol MailgunProvider: Service {
     var apiKey: String { get }
-    var customURL: String { get }
-    func sendMail(data content: MailgunEngine.FormData, on req: Request) throws -> Future<Response>
+    var domain: String { get }
+    func send(_ content: Mailgun.Message, on req: Request) throws -> Future<Response>
 }
 
-public struct MailgunEngine: MailgunProvider {
+// MARK: - Engine
+
+public struct Mailgun: MailgunProvider {
     
-    public enum MailgunError: Error {
+    public enum Problem: Error {
         case encodingProblem
+        case authenticationFailed
+        case unableToSendEmail
     }
     
-    public struct FormData: Content {
-        public static let defaultMediaType: MediaType = MediaType.urlEncodedForm
+    public struct Message: Content {
+        
+        public typealias FullEmail = (email: String, name: String?)
         
-        let from: String
-        let to: String
-        let subject: String
-        let text: String
+        public let from: String
+        public let to: String
+        public let cc: String?
+        public let bcc: String?
+        public let subject: String
+        public let text: String
+        public let html: String?
         
-        public init(from: String, to: String, subject: String, text: String) {
+        public init(from: String, to: String, cc: String? = nil, bcc: String? = nil, subject: String, text: String, html: String? = nil, attachment: [Data] = []) {
```

Comment:
It looks like nothing is being done with `attachment`. Intentional?

Reply:
yeah, I am unable to get it to work somehow ... even though the core of the thing is the same ... I am getting messages like `from` is missing and not sure if it's just a bug in the latest RC :(